gym/vector/sync_vector_env.py

```diff
@@ -164,8 +136,8 @@ def step_wait(self):
                 info,
             ) = step_api_compatibility(env.step(action), True)
             if self._terminateds[i] or self._truncateds[i]:
+                observation, info = env.reset()
```

Comment:
Is this a mistake? In the old version, reset is after updating the info dict, now it's before, so I think it will record an incorrect final_observation

Reply:
Yeah this seems to be either a mistake or a merge error, so I'm reverting it.

Reply:
This seems like a mistake rather than a merge error, and I also found the mistake in the asyn vector env. Tests break when I switch the order of these statements to the correct order, so I need to examine why the tests are wrong.

Reply:
I addressed this and now tests are passing. One issues with a naive swap of the order is that calling `obs, info = env,reset()` overwrites the `info` object where "final_observation" was saved. So I used a temp variable to write the observation from the last call to `.step()` into the `info` returned by the call to `.reset()`